s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -221,6 +220,24 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
     }
   }
 
+  private def computeMetaData(headers: Seq[HttpHeader], entity: ResponseEntity): ObjectMetadata =
+    ObjectMetadata(
+      headers ++
+      Seq(
+        `Content-Length`(entity.contentLengthOption.getOrElse(0)),
+        CustomContentTypeHeader(entity.contentType)
+      )
+    )
+
+  //`Content-Type` header is by design not accessible as header. So need to have a custom
+  //header implementation to expose that
```

Comment:
Jup. I can see how it is reasonable to construct your own Content-Type header in this case... we might want to introduce a nicer escape hatch for that at some point - though still making it evident that for 'typical' situations you should not construct it yourself.